tests/client_suite.go

```diff
@@ -1314,11 +1314,16 @@ func (s *ClientFunctionalSuite) Test_BufferedSignalCausesUnhandledCommandAndSche
 
 // Analogous to Test_BufferedSignalCausesUnhandledCommandAndSchedulesNewTask
 //
-// 1. The worker starts executing the first WFT, before any update is sent.
-// 2. While the first WFT is being executed, an update is sent.
-// 3. Once the server has received the update, the workflow completes.
-// 4. The server fails the complete request because there is an admitted update and schedules a new workflow task.
-// 5. This time the workflow accepts the update and completes the workflow.
+//  1. The worker starts executing the first WFT, before any update is sent.
+//  2. While the first WFT is being executed, an update is sent.
+//  3. Once the server has received the update, the workflow tries to complete itself.
+//  4. The server fails the complete request (and WFT) because there is an admitted update,
+//     clears workflow context, mutable state, and update registry and schedules a new workflow task.
+//  5. Now there is a race:
+//     - worker completes WFT w/o updates in it and updates in the registry,
+//     - history handler retries UpdateWorkflowExecution call and recreates update in the registry.
+//  6. In first case, workflow completes successfully after 2nd attempt, and call to UpdateWorkflowExecution
+//     returns "workflow execution already completed" error. This is what this test asserts.
```

Comment:
Given that there is a race, why is the test deterministic? Let's add explanation of this to the comment.